### PR TITLE
wxGUI/Data catalog: distinguish label and name of nodes

### PR DESCRIPTION
**Pull request by lindakladivova**
wxGUI/datacatalog: node.label changed to node.data['name'] in tree.py.

### CODE REVIEW

**Comment by petrasovaa** [comment]
You need to rebase and solve conflict with master. When you are in this PR branch do something like (always read git status to check):
```
git fetch upstream
git rebase upstream/master
# resolve conflicts in tree.py
git add tree.py
git rebase --continue
git push -f
````

**Comment by petrasovaa** [comment]
You don't need to reload the entire tree to refresh the one layer.

File: gui/wxpython/datacatalog/tree.py [hunk 1]
```diff
@@ -645,34 +645,40 @@ def OnCopyMap(self, event):
...
+                _('New name'),
+                _('Rename map'),
+                old_name,
+                env=env,
+                mapset=self.selected_mapset[0].data['name'],
+                element=self.selected_layer[0].data['type'])
+            if new_name:
+                self.Rename(old_name, new_name)
+                self.ReloadTreeItems()
+        except Exception as e:
```
Comment (petrasovaa) [inline gui/wxpython/datacatalog/tree.py:669]: Which exception are you trying to catch here? Raised in _getNewMapName or Rename?

**Comment by lindakladivova** [comment]
> You don't need to reload the entire tree to refresh the one layer.

I have tried to use self.RefreshNode(self.selected_layer[0]) instead of self.ReloadTreeItems but it does not work. 
So, I should probably make the function self.ReloadSelectedLayer similar to self.ReloadCurrentMapset which would reload only selected layer. And similarly I could make other functions like self.ReloadSelectedMapset etc.. which might be used in other rename/delete functions further on.
But I am not sure how to make self.ReloadSelectedLayer since there is some process:
p = Process(
            target=getLocationTree,
            args=(
                self.current_grassdb_node.data['name'],
                self.current_location_node.data['name'],
                q,
                self.current_mapset_node.data['name']))...
in which a layer does not appear.

**Comment by petrasovaa** [comment]
> > You don't need to reload the entire tree to refresh the one layer.
> 
> I have tried to use self.RefreshNode(self.selected_layer[0]) instead of self.ReloadTreeItems but it does not work.
> So, I should probably make the function self.ReloadSelectedLayer similar to self.ReloadCurrentMapset which would reload only selected layer. And similarly I could make other functions like self.ReloadSelectedMapset etc.. which might be used in other rename/delete functions further on.
> But I am not sure how to make self.ReloadSelectedLayer since there is some process:
> p = Process(
> target=getLocationTree,
> args=(
> self.current_grassdb_node.data['name'],
> self.current_location_node.data['name'],
> q,
> self.current_mapset_node.data['name']))...
> in which a layer does not appear.

Reloading single layer is different than mapset/location/db. For this PR just change the node.label. We will address this in future differently.

**Comment by lindakladivova** [comment]
I had an idea to changed ReloadTreeItems() in create/rename/delete functions to self.Refresh(self.selected_"node"[0], recursive=True) but it does not work. Seems to me that refreshing is different from reloading and for reload the function _initTreeItems is needed.

**Comment by petrasovaa** [comment]
> I had an idea to changed ReloadTreeItems() in create/rename/delete functions to self.Refresh(self.selected_"node"[0], recursive=True) but it does not work. Seems to me that refreshing is different from reloading and for reload, the function _initTreeItems is needed.

Yes, that's different. I suggest to review in depth the implementation of TreeModel and TreeView and AbstractTreeViewMixin. Basically, refreshing just influences rendering of the nodes, reloading changes the actual underlying data model.

File: gui/wxpython/datacatalog/tree.py (hunk 1 above)
Comment (petrasovaa) [reply gui/wxpython/datacatalog/tree.py:669]: Remove it or explain why it needs to be there.

**Pull request merged**